### PR TITLE
Revert "fix: pass `defaultValue` in to textarea and text input initial state"

### PR DESCRIPTION
**Pull request by emyarod**
Reverts carbon-design-system/carbon#3015 also reverts #2745 as it introduced state to the fully controlled `<TextInput>` which caused breaking changes

### CODE REVIEW

**Review by asudoh — COMMENTED** [review]
LGTM 👍 - Can merge once @shixiedesign agree with this revert. Thanks @emyarod!

**Review by asudoh — APPROVED** [review]
Got a green light from @shixiedesign offline - Thanks @shixiedesign!

**Review by joshblack — APPROVED** [review]

**Pull request merged**
